apps/backend/src/lib/payments.tsx

```diff
@@ -433,7 +433,7 @@ export async function validatePurchaseSession(options: {
     },
   });
 
-  if (codeData.offerId && existingOneTimePurchases.some((p) => p.offerId === codeData.offerId)) {
+  if (codeData.offerId && existingOneTimePurchases.some((p) => p.offerId === codeData.offerId && offer.stackable !== true)) {
```

Comment:
why do we still have this at all? why would the logic be different for one-time purchase offers and repeating subscriptions?